docs/source/how-to/selecting-the-collection-backend.rst

```diff
@@ -0,0 +1,131 @@
+Selecting the collection backend
+================================
+
+
+**Warning**: Backend-library disptaching at the collection level is still an experimental feature. Both the ``DaskBackendEntrypoint`` API and the set of "dispatchable" functions are expected to change.
+
+
+Changing the default backend library
+------------------------------------
+
+The Dask-Dataframe and Dask-Array modules were originally designed with the Pandas and Numpy backend libraries in mind, respectively. However, other dataframe and array libraries can take advantage of the same collection APIs for out-of-core and parallel processing. For example, users with `cupy <https://cupy.dev/>` installed can change their default Dask-Array backend to ``cupy`` with the ``"array.backend.library"`` configuration option:
+
+.. code-block:: python
+
+   >>> import dask
+   >>> import dask.array as da
+   >>> with dask.config.set({"array.backend.library": "cupy"}):
+   ...     darr = da.ones(10, chunks=(5,))  # Get cupy-backed collection
+
+This code opts out of the default (``"numpy"``) backend for dispatchable Dask-Array creation functions, and uses the creation functions registered for ``"cupy"`` instead. The current set of dispatchable creation functions for Dask-Array include:
+
+- ``ones``
+- ``zeros``
+- ``empty``
+- ``full``
+- ``arange``
+
+The Dask-Array API can also dispatch the backend ``RandomState`` class to be used for random-number generation. This means all creation functions in ``dask.array.random`` are also dispatchable.
+
+The current set of dispatchable creation functions for Dask-Dataframe include:
+
+- ``from_dict``
+- ``read_parquet``
+- ``read_json``
+- ``read_orc``
+- ``read_csv``
+- ``read_hdf``
+
+As the backend-library disptaching system becomes more mature, this set of dispatchable creation functions is likely to grow.
+
+
+Defining a new collection backend
+---------------------------------
+
+**Warning**: Defining a custom backend is **not** yet recommended for most users and down-stream libraries. The backend-entrypoint system should still be treated as experimental.
```

Comment:
xref #9508

Reply:
To expand a bit: this would be a great thing to put an experimental decorator on! Not necessarily in this PR, but putting this bookmark down.